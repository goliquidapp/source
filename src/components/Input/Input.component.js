import React, {Component} from 'react';
import { TextInput, View, Text, TouchableOpacity} from 'react-native';
import {Icon} from 'react-native-elements';
import Colors from '../../resources/Colors.js';
import Theme from '../../resources/Theme.js';

import * as Animatable from 'react-native-animatable';

class Input extends Component{
    state={minus:false}
    minusToggle=()=>{
        this.setState({minus:!this.state.minus},()=>{
            if (this.state.minus && this.props.value[0]!=='-') this.onChangeText('-'+this.props.value)
            else if (!this.state.minus && this.props.value[0]==='-') this.onChangeText(this.props.value.substr(1))
        })
    }
    onChangeText=(value)=>{
        if (this.props.withMinus){
            var minus;
            if (value[0]==='-') minus=true
            else minus=false
            
            if (this.state.minus!==minus) this.setState({minus})
        }
        this.props.onChangeText(value);
    }
	render(){
        const {minus}=this.state;
		const {readOnly,keyboardType,textStyle, containerStyle,placeholder,value,error,onChangeText, nextRef, placeholderTextColor, underline, secureTextEntry, withMinus}=this.props
		return (
			<View style={[styles.container, containerStyle]}>
				<Animatable.Text transition="opacity" useNativeDriver style={value?styles.show:styles.hide}>{placeholder}</Animatable.Text>
				<TextInput  readOnly={readOnly} 
							keyboardType={keyboardType} 
							style={[styles.input,textStyle,underline?{borderBottomWidth:1}:{borderBottomWidth:0},error?styles.error:{}]} 
							placeholder={placeholder} 
							placeholderTextColor={placeholderTextColor}
							value={value} 
                            onChangeText={withMinus?this.onChangeText:onChangeText}
							undeline={underline}
							secureTextEntry={secureTextEntry}
							ref={(ref)=>this.input=ref}
							onSubmitEditing={() => {if (nextRef) nextRef.input.focus()}}
							blurOnSubmit={ !nextRef }
							returnKeyType={ nextRef?"next":'done' }/>
                {withMinus&&<TouchableOpacity style={styles.icon} onPress={this.minusToggle}><Icon name={"minus"} type="entypo" color={minus?Theme['dark'].primaryText:Theme['dark'].secondaryText} size={18} /></TouchableOpacity>}
			</View>
		)
	}
}

Input.defaultProps={
	underline:true
}

const styles={
	container:{
		width:'90%',
		marginVertical:8
	},
	input:{
		paddingVertical:15,
		paddingHorizontal:15,
		paddingRight:40,
		borderBottomWidth:1,
		borderBottomColor:Colors['White'],
		fontSize:14,
		width:'100%',
		fontFamily:Theme['dark'].fontNormal
	},
	error:{
		borderBottomColor:Colors['Red']
	},
	show:{
		opacity:1,
		fontSize:12,
		color:Colors['Hit Gray']
	},
	hide:{
		opacity:0,
		fontSize:12,
		color:Colors['Hit Gray']
	},
    icon:{
    position:'absolute',
    right:0,
    bottom:10
    }
}

export default Input;
