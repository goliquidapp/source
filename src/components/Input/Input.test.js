import React from 'react';
import { Text, View} from 'react-native';
import renderer from 'react-test-renderer';

import Input from './Input.component';

describe("Testing Input component",()=>{
	it('renders correctly', () => {
		const wrapper = renderer.create(
    		<Input value={""}/>
    	).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
})