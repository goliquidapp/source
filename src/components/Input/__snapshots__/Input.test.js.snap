// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginVertical": 8,
        "width": "90%",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#A1ADB5",
        "fontSize": 12,
        "opacity": 0,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    blurOnSubmit={true}
    onSubmitEditing={[Function]}
    rejectResponderTermination={true}
    returnKeyType="done"
    style={
      Array [
        Object {
          "borderBottomColor": "#FFFFFF",
          "borderBottomWidth": 1,
          "fontFamily": "System",
          "fontSize": 14,
          "paddingHorizontal": 15,
          "paddingRight": 40,
          "paddingVertical": 15,
          "width": "100%",
        },
        undefined,
        Object {
          "borderBottomWidth": 1,
        },
        Object {},
      ]
    }
    undeline={true}
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
